### PR TITLE
Jennyf/automation app

### PR DESCRIPTION
**Pull request by jennyf19**
I had to put setters on the properties in AuthenticationResult in order to serialize the json response. 
Only functionality is AcquireToken, AcquireTokenSilent, and ExpireAccessToken

### CODE REVIEW

File: AutomationApp/AutomationUI.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            Logger.Callback = _loggerCallback;
+        }
+
+        public Dictionary<string, string> CreateDictionaryFromJson(string json)
+        {
+            var jss = new JavaScriptSerializer();
+            return jss.Deserialize<Dictionary<string, string>>(json);
+        }
+
+        private void AutomationApp_Load(object sender, EventArgs e)
```
Comment (kpanwar) [inline AutomationApp/AutomationUI.cs]: remove unimplemented function

File: AutomationApp/LoggerCallbackImpl.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+
+namespace AutomationApp
+{
+    class LoggerCallbackImpl : ILoggerCallback
+    {
+        private readonly StringBuilder _logCollector = new StringBuilder();
+
+        public void Log(Logger.LogLevel level, string message, bool containsPii)
```
Comment (kpanwar) [inline AutomationApp/LoggerCallbackImpl.cs]: if you have "using Microsoft.Identity.Client" then you should have no need to write Logger.LogLevel. you cans imply reference LogLevel

File: AutomationApp/AutomationUI.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
+﻿using System;
```
Comment (kpanwar) [inline AutomationApp/AutomationUI.cs]: Add MIT license header to all source files.

File: AutomationApp/TokenCacheDelegates.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+
+namespace AutomationApp
+{
+    internal static class TokenCacheDelegates
+    {
+        public static string CacheFilePath = System.Reflection.Assembly.GetExecutingAssembly().Location +
+                                             "msal-cach.txt";
```
Comment (kpanwar) [inline AutomationApp/TokenCacheDelegates.cs]: you mean "msal-cache"?

File: AutomationApp/TokenHandler.cs [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        static TokenHandler()
+        {
+            JsonLabelReplacements.Add("AccessTokenType", "access_token_type");
+            JsonLabelReplacements.Add("AccessToken", "access_token");
+            JsonLabelReplacements.Add("ExpiresOn", "expires_on");
+            JsonLabelReplacements.Add("ExtendedExpiresOn", "extended_expires_on");
+            JsonLabelReplacements.Add("ExtendedLifeTimeToken", "extended_lifetime_token");
+            JsonLabelReplacements.Add("IdToken", "id_token");
+            JsonLabelReplacements.Add("TenantId", "tenant_id");
+            JsonLabelReplacements.Add("UserInfo", "user_info");
```
Comment (kpanwar) [inline AutomationApp/TokenHandler.cs]: there is no user_info returned. please check what values are being returned and update this list accordingly.

File: AutomationApp/TokenHandler.cs [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            JsonLabelReplacements.Add("FamilyName", "family_name");
+            JsonLabelReplacements.Add("GivenName", "given_name");
+            JsonLabelReplacements.Add("IdentityProvider", "identity_provider");
+            JsonLabelReplacements.Add("PasswordChangeUrl", "password_change_url");
+            JsonLabelReplacements.Add("PasswordExpiresOn", "password_expires_on");
+            JsonLabelReplacements.Add("UniqueId", "unique_id");
+        }
+        #endregion
+
+        private static string ToJson(object obj)
```
Comment (kpanwar) [inline AutomationApp/TokenHandler.cs]: MSAL already has JsonHelper class that does the same thing as ToJson(). Consider using that instead

File: src/Microsoft.Identity.Client/AuthenticationResult.cs [hunk 7]
```diff
@@ -48,35 +50,57 @@ internal AuthenticationResult(AccessTokenCacheItem accessTokenCacheItem, Request
         /// <summary>
         /// Gets the Access Token requested.
         /// </summary>
-        public string AccessToken => _accessTokenCacheItem.AccessToken;
+        [DataMember]
+        public string AccessToken
+        {
+            get { return _accessTokenCacheItem.AccessToken; }
+            set =>  _accessTokenCacheItem.AccessToken = value;
```
Comment (kpanwar) [inline src/Microsoft.Identity.Client/AuthenticationResult.cs]: please do not change any contracts on the public methods. there should be no setter property on any of the attributes in the class.

File: src/Microsoft.Identity.Client/Internal/Requests/BaseRequest.cs [hunk 8]
```diff
@@ -45,7 +45,7 @@ internal abstract class BaseRequest
         protected TokenResponse Response;
         protected AccessTokenCacheItem AccessTokenItem;
 
-        internal RequestContext RequestContext { get; set; }
+        internal RequestContext RequestContext { get; set; } = new RequestContext(Guid.Empty);
```
Comment (kpanwar) [inline src/Microsoft.Identity.Client/Internal/Requests/BaseRequest.cs]: this should never be set to Guid.Empty. It will always be created in public and confidential client class

File: src/Microsoft.Identity.Client/Platforms/net45/WindowsFormsWebAuthenticationDialogBase.cs [hunk 9]
```diff
@@ -42,13 +42,8 @@ namespace Microsoft.Identity.Client
     [EditorBrowsable(EditorBrowsableState.Never)]
     public abstract class WindowsFormsWebAuthenticationDialogBase : Form
     {
-        private RequestContext RequestContext { get; }
+        private RequestContext RequestContext { get; set; } = new RequestContext(Guid.Empty);
```
Comment (kpanwar) [inline src/Microsoft.Identity.Client/Platforms/net45/WindowsFormsWebAuthenticationDialogBase.cs]: this should not be initialized to Guid.Empty. update the code to call WindowsFormsWebAuthenticationDialogBase(RequestContext requestContext) instead. you could make the class and copnstructor internal

File: tests/Test.MSAL.NET.Unit/Test.MSAL.NET.Unit.csproj [hunk 10]
```diff
@@ -90,4 +90,7 @@
     <PackageReference Include="MSTest.TestAdapter" Version="1.1.11" />
     <PackageReference Include="MSTest.TestFramework" Version="1.1.11" />
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
```
Comment (kpanwar) [inline tests/Test.MSAL.NET.Unit/Test.MSAL.NET.Unit.csproj]: why is this change here?

File: AutomationApp/AutomationUI.cs (hunk 3 above)
Comment (jennyf19) [inline AutomationApp/AutomationUI.cs]: Done.

File: AutomationApp/AutomationUI.cs (hunk 1 above)
Comment (jennyf19) [inline AutomationApp/AutomationUI.cs]: Done.

File: AutomationApp/LoggerCallbackImpl.cs (hunk 2 above)
Comment (jennyf19) [inline AutomationApp/LoggerCallbackImpl.cs]: The LogLevel is an enum inside of the Logger class, which is why I need to use the qualified name. If there is a different way to reference it, let me know.

File: AutomationApp/TokenCacheDelegates.cs (hunk 4 above)
Comment (jennyf19) [inline AutomationApp/TokenCacheDelegates.cs]: good catch

File: AutomationApp/TokenHandler.cs (hunk 5 above)
Comment (jennyf19) [inline AutomationApp/TokenHandler.cs]: Done.

File: AutomationApp/TokenHandler.cs (hunk 6 above)
Comment (jennyf19) [inline AutomationApp/TokenHandler.cs]: Done.

File: src/Microsoft.Identity.Client/AuthenticationResult.cs (hunk 7 above)
Comment (jennyf19) [inline src/Microsoft.Identity.Client/AuthenticationResult.cs]: Serialization requires properties to have setters. Suggestions?

File: src/Microsoft.Identity.Client/Internal/Requests/BaseRequest.cs (hunk 8 above)
Comment (jennyf19) [inline src/Microsoft.Identity.Client/Internal/Requests/BaseRequest.cs]: I noticed that when there is no guid present, the application will blow up. I'm using defensive coding, to protect the app against this edge case. The request object will be overwritten with the correct guid when it's set on the BaseRequest object.

File: src/Microsoft.Identity.Client/Platforms/net45/WindowsFormsWebAuthenticationDialogBase.cs (hunk 9 above)
Comment (jennyf19) [inline src/Microsoft.Identity.Client/Platforms/net45/WindowsFormsWebAuthenticationDialogBase.cs]: I noticed there is a ComVisible attribute on the class. Do you still want this to be internal? It will go against the ComVisible setting.

File: tests/Test.MSAL.NET.Unit/Test.MSAL.NET.Unit.csproj (hunk 10 above)
Comment (jennyf19) [inline tests/Test.MSAL.NET.Unit/Test.MSAL.NET.Unit.csproj]: Removed.
